### PR TITLE
feat(cli): add cli option to specify working path

### PR DESCRIPTION
**Pull request by finswimmer**
# Pull Request Check List

Resolves: #2179, https://github.com/python-poetry/poetry/issues/7039

- [ ] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by finswimmer** [comment]
Could need some help to write proper tests :pray:

**Review by lovelyyfiaaa — COMMENTED** [review]
LGTM! It's pretty small sized!

Yupp, we might want some test here, but I'll assume that to be simple!

File: docs/cli.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ then `--help` combined with any of those can give you more information.
 * `--no-interaction (-n)`: Do not ask any interactive question.
 * `--no-plugins`: Disables plugins.
 * `--no-cache`: Disables Poetry source caches.
+* `--project=PROJECT`: Specify path to Poetry project the command should work at.
```
Comment (neersighted) [inline docs/cli.md:32]:
```suggestion
* `--project=PATH`: The working directory for the Poetry command (defaults to the current working directory).
```

File: src/poetry/console/application.py [hunk 2]
```diff
@@ -367,6 +372,16 @@ def _default_definition(self) -> Definition:
             )
         )
 
+        definition.add_option(
+            Option(
+                "--project",
+                flag=False,
+                description=(
+                    "Specify path to Poetry project the command should work at."
```
Comment (neersighted) [inline src/poetry/console/application.py:380]: As above (and maybe we should consider a short flag like Git's `-C`?).

File: src/poetry/console/commands/init.py [hunk 3]
```diff
@@ -77,7 +77,18 @@ def handle(self) -> int:
...
 
-        pyproject = PyProjectTOML(Path.cwd() / "pyproject.toml")
+        project_path = Path.cwd()
+
+        if self.io.input.option("project"):
+            project_path = Path(self.io.input.option("project"))
+            if not project_path.exists() or not project_path.is_dir():
+                self.line_error(
+                    "<error>The given target for the project does not exists or is not"
+                    " a folder.</error>"
```
Comment (neersighted) [inline src/poetry/console/commands/init.py:87]:
```suggestion
                    "<error>The --project path is not a directory.</error>"
```

File: src/poetry/console/commands/new.py [hunk 4]
```diff
@@ -34,6 +34,13 @@ def handle(self) -> int:
         from poetry.layouts import layout
         from poetry.utils.env import SystemEnv
 
+        if self.io.input.option("project"):
+            self.line_error(
+                "<warning>--project can only be used on existing projects. Poetry will"
+                " ignore this option. Use --path instead to define the path to create"
+                " the project at.</warning>"
```
Comment (neersighted) [inline src/poetry/console/commands/new.py:41]:
```suggestion
                "<warning>--project can only be used on existing projects. Poetry will"
                "<warning>--project only makes sense with existing projects, and will"
                " be ignored. You should consider the option --path instead.</warning>"
```

**Review by neersighted — CHANGES_REQUESTED** [review]
Some phrasing changes, and I'd like to consider a short flag ala `-C`.

File: src/poetry/console/application.py (hunk 2 above)
Comment (finswimmer) [reply src/poetry/console/application.py]: I'm happy to add a short flag. But `-C` doesn't seem intuitive. Maybe `-p`? :thinking:

**Comment by clintonroy** [comment]
-C, --directory are the common options for this in unix tools (make, tar)  if you're willing to tweak the doc a tiny bit to match that.

**Comment by finswimmer** [comment]
> -C, --directory are the common options for this in unix tools (make, tar) if you're willing to tweak the doc a tiny bit to match that.

Fair enough. But https://xkcd.com/1168/ :rofl:

**Review by neersighted — CHANGES_REQUESTED** [review]
I'd like to see this as `--directory, -C` only, as not only does it match other tools, but `--project` feels more vague in intent/subject to misinterpretation. 

The implementation LGTM.

**Review by neersighted — APPROVED** [review]

**Pull request merged**